### ISSUE TITLE
Markdown Inline Image Issue

### ISSUE THREAD

**Issue by zchrykng** [sublimehq/Packages#103]
Not sure if this has already been reported.

Inline image links appear to be having issues with syntax highlighting. I tried the Markdown Extended which highlights correctly, but spell checking marks it wrong with either syntax option.

![Screenshot](http://d.pr/i/11p9b+)

**Comment by kleinfreund** [comment]
Is the highlighting not off by one letter as well?

**Comment by philipbordallo** [comment]
Changing the `3:` on [line 283][1] to a `4:` fixes the highlighting issue, but doesn't seem to fix the spell checking issue.

[1]: https://github.com/sublimehq/Packages/blob/f7f96a0c31f6d6362dcb7ebfab6d1a0db7641f60/Markdown/Markdown.sublime-syntax#L283

**Comment by wbond** [comment]
The spelling issue can be solved by making the character set `[^\[\]\\]` on https://github.com/sublimehq/Packages/blob/f7f96a0c31f6d6362dcb7ebfab6d1a0db7641f60/Markdown/Markdown.sublime-syntax#L266 match one or more characters: `[^\[\]\\]+`.

However, I believe this would just result in hiding a subtle bug. It appears as if the final capture value for group 3 is not being passed with the rest of the capture value from capture 2 into the spell checker.

**Comment by zchrykng** [comment]
Sounds about right to me. I don't know much about the syntax definitions, but if I have time later I may try to take a look at this again.

**Comment by wbond** [comment]
I looked into this a little more and determined the issue seems to stem from the use of a named capture group to handle the `[]` recursion. I'll post more if I figure out a way to deal with it, however it may end up being a limitation of the oniguruma regex engine.

**Comment by wbond** [comment]
The incorrect highlighting was fixed by e3c6bb8969ccb820c2f282be08d238f96dceab4c.

The spelling issue had to do with how ST tokenizes. Since capture groups are used to apply scopes, the capture of the last letter was creating a separate token from the rest of the description. This resulted in the word being composed to two tokens, and the spell checker only runs on a single token at a time.

Because the capture groups need to create separate tokens for the sake of applying different scope names, there isn't an automatic way to solve this kind of issue.

I fixed it for Markdown with 35d1a20fbf18ba5948cd1c8d895a22fb27ce619f by adding repetition inside of the capture group. This results in a single token, and thus the spell checker can properly identify the word.
